addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlHandler.java

```diff
@@ -0,0 +1,224 @@
+                    int stepValue = (Integer) thing.getConfiguration().get(CONFIG_STEP_VALUE);
+                    int currentValue = nhcComm.getActionState(actionID);
+                    int newValue;
+                    if (s == IncreaseDecreaseType.INCREASE) {
+                        newValue = currentValue + stepValue;
+                        // round down to nearest step multiple
+                        newValue = newValue - newValue % stepValue;
+                        nhcComm.executeAction(actionID, (newValue > 100 ? 100 : newValue));
+                    } else {
+                        newValue = currentValue - stepValue;
+                        // round up to nearest step multiple
+                        newValue = newValue + newValue % stepValue;
+                        nhcComm.executeAction(actionID, (newValue < 0 ? 0 : newValue));
+                    }
+                } else if (command instanceof PercentType) {
+                    PercentType p = (PercentType) command;
+                    nhcComm.executeAction(actionID, p.intValue());
+                }
+                updateStatus(ThingStatus.ONLINE);
+                break;
+
+            case CHANNEL_ROLLERSHUTTER:
+                if (command instanceof UpDownType) {
+                    UpDownType s = (UpDownType) command;
+                    if (s == UpDownType.UP) {
+                        nhcComm.executeAction(actionID, 255);
+                        logger.debug("Niko Home Control: rollershutter {} up pressed", actionID);
+                    } else {
+                        nhcComm.executeAction(actionID, 254);
+                        logger.debug("Niko Home Control: rollershutter {} down pressed", actionID);
+                    }
+                } else if (command instanceof StopMoveType) {
+                    StopMoveType s = (StopMoveType) command;
+                    if (s == StopMoveType.STOP) {
+                        nhcComm.executeAction(actionID, 253);
+                        logger.debug("Niko Home Control: rollershutter {} stop pressed", actionID);
+                    } else {
+                        logger.debug("Niko Home Control: rollershutter {} move pressed", actionID);
+                    }
+                } else if (command instanceof PercentType) {
+                    PercentType p = (PercentType) command;
+                    nhcComm.executeAction(actionID, p.intValue());
+                    logger.debug("Niko Home Control: rollershutter {} percent {}", actionID, p.intValue());
+                }
+                updateStatus(ThingStatus.ONLINE);
+                break;
+
+            default:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Niko Home Control: channel unknown " + channelUID.getId());
+        }
+    }
+
+    @Override
+    public void initialize() {
+
+        int actionID;
```

Comment:
You can put things in the configuration during discovery, although its API is a little unexpected.

All configuration parameters which are defined can just be set on discovery by calling `setProperty` / `setProperties`. Every 'property' which is not known as a configuration parameter will be stored as property the rest as configuration.

Does this help you?